src/app/webui/components/ToolConfirmationHandler.tsx

```diff
@@ -0,0 +1,191 @@
+'use client';
+
+import React, { useEffect, useState, useCallback } from 'react';
+import { Dialog, DialogContent, DialogHeader, DialogTitle, DialogFooter } from './ui/dialog';
+import { Button } from './ui/button';
+import { Checkbox } from './ui/checkbox';
+import { Badge } from './ui/badge';
+import { AlertTriangle, Wrench, Clock } from 'lucide-react';
+
+interface ToolConfirmationEvent {
+    executionId: string;
+    toolName: string;
+    args: any;
+    description?: string;
+    timestamp: Date;
+}
+
+interface ToolConfirmationHandlerProps {
+    websocket?: WebSocket | null;
+}
+
+/**
+ * WebUI component for handling tool confirmation requests
+ * Displays confirmation dialogs and sends responses back through WebSocket
+ */
+export function ToolConfirmationHandler({ websocket }: ToolConfirmationHandlerProps) {
+    const [pendingConfirmation, setPendingConfirmation] = useState<ToolConfirmationEvent | null>(null);
+    const [rememberChoice, setRememberChoice] = useState(false);
+    const [isDialogOpen, setIsDialogOpen] = useState(false);
+
+    // Handle incoming WebSocket messages
+    useEffect(() => {
+        if (!websocket) return;
+
+        const handleMessage = (event: MessageEvent) => {
+            try {
+                const message = JSON.parse(event.data);
+                
+                if (message.event === 'toolConfirmationRequest') {
+                    const confirmationEvent: ToolConfirmationEvent = {
+                        ...message.data,
+                        timestamp: new Date(message.data.timestamp)
+                    };
+                    
+                    setPendingConfirmation(confirmationEvent);
+                    setIsDialogOpen(true);
+                    setRememberChoice(false);
+                }
+            } catch (error) {
+                console.error('Error parsing WebSocket message:', error);
+            }
+        };
+
+        websocket.addEventListener('message', handleMessage);
+        
+        return () => {
+            websocket.removeEventListener('message', handleMessage);
+        };
+    }, [websocket]);
+
+    // Send confirmation response
+    const sendResponse = useCallback((approved: boolean) => {
+        if (!pendingConfirmation || !websocket) return;
+
+        const response = {
+            type: 'toolConfirmationResponse',
+            data: {
+                executionId: pendingConfirmation.executionId,
+                approved,
+                rememberChoice,
+            }
+        };
```

Comment:
we probably need documentation for this somewhere

and maybe examples for how to use this